include/openPMD/backend/Attributable.hpp

```diff
@@ -90,21 +90,37 @@ namespace internal
         A_MAP m_attributes;
     };
 
+    enum class SetAttributeMode : char
+    {
+        WhileReadingAttributes,
+        FromPublicAPICall
```

Comment:
Is that public API call both read and write? Might need a doxygen string, even though its readable already :)

Reply:
Ah, `Set`... so write